USWDS -Site - Layout Grid: Guidance refactor (#2276)
uswds/uswds-site: _utilities/layout-grid.md

Reviewer: ## Question (nitpick, if-minor)

It looks like some of headings and text wrapping changed quite a bit. Was that intentional?

FWIW, neither flagged a11y issues in axe devtools.

**Feature branch** (left) / **Production** (right)
![image](https://github.com/uswds/uswds-site/assets/3385219/74b895d4-6199-4ef1-9568-954d9cce22bb)

![image](https://github.com/uswds/uswds-site/assets/3385219/9c368660-1e37-4fd9-96b7-a0f79d474cce)

![image](https://github.com/uswds/uswds-site/assets/3385219/5280c1d8-9932-4ada-949e-126ae12a4957)

![image](https://github.com/uswds/uswds-site/assets/3385219/a67eecb1-3ddd-4aa1-812a-df88f052d01f)

Author: This was a result of removing `type: docs` from the frontmatter. This simultaneously removes the `usa-prose site-prose` classes from the content wrapper div and these headings take the `styleguide-content` class styles that were being overridden.

Comparing to other utilities, it looks like this is consistent across the other pages.

Author: I agree! Resolved in [ed1db94](https://github.com/uswds/uswds-site/pull/2276/commits/ed1db94e4d86fb679bbfc30c50161d2e3380c789)

Author: Noting that for the `grid-col-fill` section, I made one example class and text `grid-col` and the other `grid-col-fill` to demonstrate that both allow the column to stretch to fill available space. This matches the documentation directly above it:

![image](https://github.com/user-attachments/assets/da859763-6e69-44b4-8524-2e696c9422f5)

**Related code:**

```scss
.grid-col {
    flex: 1 1 0% !important;
    min-width: 1px;
}

.grid-col-fill {
    flex: 1 1 0% !important;
    min-width: 1px;
}
```

Reviewer 2: Good idea!

  href: '#sass-mixins'
- text: Latest updates
  href: '#changelog'

> spacing_settings:

Author: I do like the idea of only maintaining this information in one spot. Originally, I tried pulling in `spacing.yml` but since it pulled in border radius settings, I opted to create a table of the relevant settings instead.

I also considered using the liquid [slice filter](https://shopify.github.io/liquid/filters/slice/) to use `spacing.yml` but without the border-radius settings. 

```md
{% assign spacing_settings = site.data.settings.spacing.contents | slice: 3, 9 %}

{% include settings-table.html
  settings=spacing_settings
%}
```

Which outputs:

![image](https://github.com/uswds/uswds-site/assets/25211650/3d192823-7afe-4a9e-81bf-eae40db0fb86)

But this seems pretty fragile as well if `spacing.yml` changes at all.

My initial goal was to replace this code block:

![image](https://github.com/uswds/uswds-site/assets/25211650/f4fb5155-f14f-4928-8668-7229bca10947)

Do you think it would be valid to change the `#spacing-settings` jump link to link to the [spacing settings](https://designsystem.digital.gov/documentation/settings/#spacing-settings) on the settings page and not have this information on this page?

> spacing_settings:

Author: I went and ahead and removed the `spacing_settings` front matter and implemented the slice method to showcase the idea but I'm not sure we'd want to rely on this solution.

https://github.com/uswds/uswds-site/blob/2872121df1a5ac2ccfafb062159c9f9f6e23329b/_utilities/layout-grid.md?plain=1#L363-L369

Take a look and let me know if we're better just removing this table and forwarding users to the settings page to find this information

> spacing_settings:

Author: Removed in [245919e](https://github.com/uswds/uswds-site/pull/2276/commits/245919e23d7546be5ac0d183d516bab90252c694)

  - name: Small grid gap
    subsection: grid-gap token
    section: true

>     description: Sets `"grid-gap-sm"` token.

Reviewer 2: I believe the token is just called "sm"

```suggestion
    description: Sets the `.grid-gap-sm` utility class and the `"sm"` grid gap token.
```

Author: Removed in [5b6e4da](https://github.com/uswds/uswds-site/pull/2276/commits/5b6e4da238ec15b95c66f14f23af8b3dab049315)

    var: $theme-column-gap-sm
    default: 2px
    type: units
  - name: Medium grid gap
    subsection: grid-gap token

>     description: Sets `"grid-gap-md"` token.

Reviewer 2: ```suggestion
    description: Sets the `.grid-gap-md` utility class and the `"md"` grid gap token.
```

Author: Removed in [5b6e4da](https://github.com/uswds/uswds-site/pull/2276/commits/5b6e4da238ec15b95c66f14f23af8b3dab049315)

    var: $theme-column-gap-md
    default: 2
    type: units
  - name: Large grid gap
    subsection: grid-gap token
    description: Sets `"grid-gap-lg"` token.

>     var: $theme-column-gap-lg

Reviewer 2: ```suggestion
    description: Sets the `.grid-gap-lg` utility class and the `"lg"` grid gap token.
```

Author: Removed in [5b6e4da](https://github.com/uswds/uswds-site/pull/2276/commits/5b6e4da238ec15b95c66f14f23af8b3dab049315)

    default: 3
    type: units
  - name: Mobile grid gap
    subsection: grid-gap token

>     description: Sets value of `"grid-gap"` at `mobile` width.

Reviewer 2: ```suggestion
    description: Sets the value of the `.grid-gap` utility class at narrow window widths.
```

Author: Removed in [5b6e4da](https://github.com/uswds/uswds-site/pull/2276/commits/5b6e4da238ec15b95c66f14f23af8b3dab049315)

    var: $theme-column-gap-mobile
    default: 2
    type: units
  - name: Desktop grid gap
    subsection: grid-gap token

>     description: Sets value of `"grid-gap"` at `desktop` width.

Reviewer 2: ```suggestion
    description: Sets the value of the `.grid-gap` utility class at `desktop` window widths.
```

Author: Removed in [5b6e4da](https://github.com/uswds/uswds-site/pull/2276/commits/5b6e4da238ec15b95c66f14f23af8b3dab049315)

- **Media queries:** Grid breakpoints are based on minimum-width media queries, meaning they apply to that specific width and all greater widths (e.g., `tablet:col-4` applies to tablet, desktop, and widescreen devices but not at `mobile-lg` or any width below the tablet breakpoint). Refer to [responsive variants](#responsive-variants) for a full list.
- **Sass mixins:** You can use predefined grid classes (like `grid-col-4`) for presentational markup or [Sass mixins](#sass-mixins) for more semantic markup.

<h2 id="responsive-variants">Responsive variants</h2>

<div>
  <table class="usa-table--borderless site-table-responsive site-table-simple">
    <caption>Default responsive sizes</caption>
    <thead>
      <tr>

>         <th scope="col" class="flex-2">Width</th>

Author: Moved under the Responsive Variants guidance on line 517

      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex">
          <span class="text-normal">
            .grid-col <br />
            .grid-col-<code>column count</code>
          </span>
        </th>
        <td data-title="Example" class="display-inline-flex">
          <span>
            @include grid-col; <br/>
            @include grid-col(6); <br/>
            @include grid-col("auto");
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex">
          <span>
            auto, fill, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12
          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex">
          <span class="text-normal">
            .grid-gap <br />
            .grid-gap-<code>gap size</code>
          </span>
        </th>
        <td data-title="Example" class="display-inline-flex">
          <span>
            @include grid-gap; <br />
            @include grid-gap("lg");
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex">
          <span>

>             sm, md, lg, 0, 2px, 05, 1, 2, 3, 4, 6

Reviewer 2: ```suggestion
            sm, md, lg, 0, 2px, 05, 1, 2, 3, 4, 5, 6
```

    <thead>
      <tr>
        <th scope="col" class="tablet:maxw-card-lg display-inline-flex flex-2">Utility</th>
        <th scope="col" class="display-inline-flex flex-2">Mixin</th>
        <th scope="col" class="display-inline-flex flex-1">Accepted values</th>
        <th scope="col" class="display-inline-flex flex-2">Example</th>
      </tr>
    </thead>
    <tbody class="font-mono-2xs">
      <tr>
        <td data-title="Utility" class="tablet:maxw-card-lg flex-2">
          <span>
            .grid-container <br />
            .grid-container-<code>container size</code>
          </span>
        </td>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>
            grid-container<code>(container size)</code>
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>
            mobile, mobile-lg, tablet, tablet-lg, desktop, desktop-lg, or widescreen
          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include grid-container <br/>

>             @include grid-container("tablet")

Author: Do we want to do this for all utility mixins? I hadn't included this originally because that wasn't the existing pattern throughout sites utility mixin guidance.

Here's [display](https://designsystem.digital.gov/utilities/display/#utility-mixins-2) mixins for example:

![image](https://github.com/user-attachments/assets/e1de3c41-b154-47ab-8ee3-2ca410c1964d)

>             @include grid-container("tablet")

Reviewer 2: I like the idea of including semi-colons on this page _in particular_ because we offer multiple `@include` examples in the same table cell. Adding semi-colons would:
1. Follow expected code convention
2. Clearly separate distinct `@include` statements in the same table cell

Unresolving this comment and added some code suggestions in my comments below.

>             @include grid-container("tablet")

Author: That makes sense to me! I considered adding it to only the lines with multiple examples but kept em all to be consistent.

Resolved in [c892bf4](https://github.com/uswds/uswds-site/pull/2276/commits/c892bf454ffbb136fab1f4fa1fa493265109aff9)

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .grid-row
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>
            grid-row
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>
            -
          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>

>             @include grid-row

Reviewer: **issue:** Please add semicolons to SCSS examples.

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .grid-col <br />
            .grid-col-<code>column count</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>
            grid-col<code>(column count)</code>
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>

>             sm, md, lg, 2px, 05, 1, 3, 4, 6

Reviewer: **issue**: The accepted values listed under grid col look incorrect.

![image](https://github.com/user-attachments/assets/a739bcfc-d1c8-475e-a3a1-d9688f8b5d62)

**Expected values**
```
auto, fill, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12
```

Author: Thanks, looks like some of these values got moved around during edits.

Resolved in https://designsystem.digital.gov/design-tokens/order/

          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include grid-col <br/>
            @include grid-col(6) <br/>

>             @include grid-col("auto")

Reviewer: **issue:** Please add semicolons to SCSS examples.

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .grid-gap <br />
            .grid-gap-<code>gap size</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>
            grid-gap(<a href="{{ site.baseurl }}/design-tokens/spacing-units/" class="token">units</a>)
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>

>             auto, fill, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12

Reviewer: **issue**: incorrect grid gap values.

```suggestion
            sm, md, lg, 2px, 05, 1, 3, 4, 6
```

Author: Resolved in https://designsystem.digital.gov/design-tokens/order/

          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include grid-gap <br />

>             @include grid-gap("lg")

Reviewer: **issue:** Please add semicolons to SCSS examples.

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .grid-offset-<code>column-count</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>
            grid-offset<code>(column-count)</code>
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>
            none, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>

>             @include grid-offset(4)

Reviewer: **issue:** Please add semicolons to SCSS examples.

          </span>
        </td>

>       </tr>

Reviewer: **issue**: Order seems to be missing.

From live example:
```
// $order can be first, last, initial, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
@include u-order($order);
```

![image](https://github.com/user-attachments/assets/f26dbcc8-e0f5-42ab-90dd-a381704b3f0b)

Author: I added this in [3102d17](https://github.com/uswds/uswds-site/pull/2276/commits/3102d174fefed7c70e8491b8a689ad3e269d4037)

But I'm curious if it's necessary or fits here. Order isn't exclusive to layout grid and we offer the [Order design token guidance](https://designsystem.digital.gov/design-tokens/order/). I'm also concerned about potential accessibility concerns given the changes we've made recently to remove visual order changes via CSS properties.

Do you think it's necessary to include here?

Reviewer: Not if it's in a place that makes more sense and the information isn't lost. Thanks.

Author: Since the guidance is available on the order design token page, and since reordering via CSS generally goes against accessibility recommendations, I went ahead and removed the order utility mixin from this table.

Resolved in [c892bf4](https://github.com/uswds/uswds-site/pull/2276/commits/c892bf454ffbb136fab1f4fa1fa493265109aff9)

  </div>
</div>

## Theme settings

> You can redefine column gap token sizes, set the grid container max width, and update site margin by defining the relevant theme setting in your `_uswds-theme.scss` file.

Reviewer 2: **issue:** Pointing users to `_uswds-theme.scss` feels incomplete and confusing. We should point to our standard settings instructions here. 
```suggestion
You can redefine column gap token sizes, set the grid container max width, and update site margin by defining the relevant theme setting in your [USWDS settings configuration]({{ site.baseurl }}/documentation/settings#configuring-custom-uswds-settings).
```

Author: Good call! Resolved in [87c040b](https://github.com/uswds/uswds-site/pull/2276/commits/87c040b478aa2922ab341db331b65a95f68c7e1d)

> Visit the [settings page]({{ site.baseurl }}/documentation/settings) for more information and available settings.

Reviewer 2: **issue** We should point users directly to the list of grid settings (or at least as close as we can). Here is a possible suggestion to work with the suggested edit above:
```suggestion
Find a full list of the available [grid settings]({{ site.baseurl }}/documentation/settings#spacing-settings) on the settings page.
```

Author: I like this. Great suggestion! Resolved in [87c040b](https://github.com/uswds/uswds-site/pull/2276/commits/87c040b478aa2922ab341db331b65a95f68c7e1d)

<section id="utility-mixins" class="padding-top-4">
  <h2 class="site-h2 margin-y-0">Utility mixins</h2>
  {% include utilities/utility-mixin-intro.html %}

  <table class="usa-table--borderless site-table-responsive site-table-simple">
    <thead>
      <tr>
        <th scope="col" class="tablet:maxw-card-lg display-inline-flex flex-2">Utility</th>
        <th scope="col" class="display-inline-flex flex-2">Mixin</th>
        <th scope="col" class="display-inline-flex flex-1">Accepted values</th>

>         <th scope="col" class="display-inline-flex flex-2">Example</th>

Author: @amyleadem This is a really good point. I originally wrapped the parenthesis in the code tags to _hopefully_ demonstrate that it was optional but I don't think that did the trick.

After looking at the other utility pages again, it looks like most rely on a tokens table with visual examples rather than an `accepted values` column in their mixin tables. 

For reference, here's how [flex guidance](https://designsystem.digital.gov/utilities/flex/) handles flex wrap which doesn't have a design token page to show available units:

![image](https://github.com/user-attachments/assets/9f08586f-9ea4-4330-924e-dc8579e1c936)

![image](https://github.com/user-attachments/assets/c5477dd8-87e3-438c-a310-9c1c72465cb2)

I originally opted to maintain the guidance at the top of the page because of the increased scope. Bringing it up in case it's something we'd like to explore down the road.

For now, I opted to follow your recommendation but named the header `mixin` to mirror the `utility` column.

Resolved in [7af5f03](https://github.com/uswds/uswds-site/pull/2276/commits/7af5f034089632d76c15e6525bc87b8b9fa49ae0)

      </tr>
    </thead>
    <tbody class="font-mono-2xs">
      <tr>
        <td data-title="Utility" class="tablet:maxw-card-lg flex-2">
          <span>
            .grid-container <br />
            .grid-container-<code>container size</code>
          </span>
        </td>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>

>             grid-container<code>(container size)</code>

Author: Column removed in [7af5f03](https://github.com/uswds/uswds-site/pull/2276/commits/7af5f034089632d76c15e6525bc87b8b9fa49ae0)

          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>

>             mobile, mobile-lg, tablet, tablet-lg, desktop, desktop-lg, or widescreen

Author: Resolved in [4ef3432](https://github.com/uswds/uswds-site/pull/2276/commits/4ef3432120e01e02108eb0ed77c7cc93575d69ee)

          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include grid-container; <br/>
            @include grid-container("tablet");
          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .grid-row
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>
            grid-row
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>
            -
          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>

>             @include grid-row

Author: Resolved in [c892bf4](https://github.com/uswds/uswds-site/pull/2276/commits/c892bf454ffbb136fab1f4fa1fa493265109aff9)

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .grid-col <br />
            .grid-col-<code>column count</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>

>             grid-col<code>(column count)</code>

Author: Column removed in [7af5f03](https://github.com/uswds/uswds-site/pull/2276/commits/7af5f034089632d76c15e6525bc87b8b9fa49ae0)

          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>
            auto, fill, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12
          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include grid-col <br/>
            @include grid-col(6) <br/>

>             @include grid-col("auto")

Author: Resolved in [c892bf4](https://github.com/uswds/uswds-site/pull/2276/commits/c892bf454ffbb136fab1f4fa1fa493265109aff9)

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .grid-gap <br />
            .grid-gap-<code>gap size</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>

>             grid-gap(<a href="{{ site.baseurl }}/design-tokens/spacing-units/" class="token">units</a>)

Author: Removed in [7af5f03](https://github.com/uswds/uswds-site/pull/2276/commits/7af5f034089632d76c15e6525bc87b8b9fa49ae0)

          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>

>             sm, md, lg, 2px, 05, 1, 3, 4, 6

Author: Resolved in [4ef3432](https://github.com/uswds/uswds-site/pull/2276/commits/4ef3432120e01e02108eb0ed77c7cc93575d69ee)

          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include grid-gap <br />

>             @include grid-gap("lg")

Author: Resolved in [c892bf4](https://github.com/uswds/uswds-site/pull/2276/commits/c892bf454ffbb136fab1f4fa1fa493265109aff9)

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">

>             .grid-offset-<code>column-count</code>

Author: Resolved in [4ef3432](https://github.com/uswds/uswds-site/pull/2276/commits/4ef3432120e01e02108eb0ed77c7cc93575d69ee)

          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>

>             grid-offset<code>(column-count)</code>

Author: Removed this column in [7af5f03](https://github.com/uswds/uswds-site/pull/2276/commits/7af5f034089632d76c15e6525bc87b8b9fa49ae0)

          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>

>             none, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11

Author: Resolved in [4ef3432](https://github.com/uswds/uswds-site/pull/2276/commits/4ef3432120e01e02108eb0ed77c7cc93575d69ee)

          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>

>             @include grid-offset(4)

Author: Resolved in [c892bf4](https://github.com/uswds/uswds-site/pull/2276/commits/c892bf454ffbb136fab1f4fa1fa493265109aff9)

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .order-<code>column-position</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>

>             u-order<code>(column-position)</code>

Author: Column removed in [7af5f03](https://github.com/uswds/uswds-site/pull/2276/commits/7af5f034089632d76c15e6525bc87b8b9fa49ae0)

          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>

>             first, last, initial, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11

Author: Resolved in [4ef3432](https://github.com/uswds/uswds-site/pull/2276/commits/4ef3432120e01e02108eb0ed77c7cc93575d69ee)

          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include u-order(4)

>             @include u-order("first")

Author: Resolved in [c892bf4](https://github.com/uswds/uswds-site/pull/2276/commits/c892bf454ffbb136fab1f4fa1fa493265109aff9)

  </div>
</div>

## Theme settings
You can redefine column gap token sizes, set the grid container max width, and update site margin by defining the relevant theme setting in your `_uswds-theme.scss` file.

> Visit the [settings page]( {{ site.baseurl }}/documentation/settings) for more information and available settings.

Author: ## Theme settings section

In an effort to replace the theme settings section:

![image](https://github.com/uswds/uswds-site/assets/25211650/408d0bf3-481a-48a8-9637-c5749e162758)

I added this basic outline of the settings available to users before pointing users to the settings page for more information. 

While other utility pages do not include this section, I saw some value in maintaining the information instead of removing it completely. If we decide we don't need it, we can remove it entirely.

Author: Removed this section

> Visit the [settings page]( {{ site.baseurl }}/documentation/settings) for more information and available settings.

Author: Resolved in [f1c0466](https://github.com/uswds/uswds-site/pull/2276/commits/f1c0466a7c6f1af2a1ef576e8489f329d69c3400)

        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include grid-gap <br />
            @include grid-gap("lg")
          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:maxw-card-lg display-inline-flex flex-2">
          <span class="text-normal">
            .grid-offset-<code>column-count</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex flex-2">
          <span>
            grid-offset<code>(column-count)</code>
          </span>
        </td>
        <td data-title="Accepted values" class="display-inline-flex flex-1">
          <span>
            none, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
          </span>
        </td>
        <td data-title="Example" class="display-inline-flex flex-2">
          <span>
            @include grid-offset(4)
          </span>
        </td>
      </tr>
    </tbody>
  </table>
  {% include utilities/utility-mixin-using.html %}
</section>  

<section id="advanced-settings" class="padding-top-4">
  <h2 class="site-h2 margin-y-0">Advanced settings</h2>

  {% include utilities/responsive-variants.html %}

>   No settings available.

Author: Created #2748 to track. 

I tagged it with `good first issue` because I thought it might be a good one for @cathybaptista or @RachelCorsino !

>   No settings available.

Reviewer: Thanks Charlie!

When generating your CSS from USWDS source files, you have the option of customizing many system defaults by modifying project theme variables. USWDS also provides grid mixins for adding grid functionality to custom semantic component CSS.

### Variables
Variables and maps determine the number of columns, gutter width, and media-query point at which to begin floating columns. We use variables to generate the predefined grid classes documented on this page, as well as for the custom mixins noted under [spacing settings](#spacing-settings) and [utility mixins](#utility-mixins).

#### Spacing settings

{% include settings-table.html
  settings=page.spacing_settings
%}

<section id="utility-mixins" class="padding-top-4">
  <h2 class="site-h2 margin-y-0">Utility mixins</h2>

>   {% include utilities/utility-mixin-intro.html %}

Reviewer 2: Now that we are adding the standard mixin intro, can we remove the "Sass mixins" section?

Reviewer: Was this resolved?

Author: Resolved in [57ef05b](https://github.com/uswds/uswds-site/pull/2276/commits/57ef05baeb9957e4c129ce108f06e3daf3357d36)

          <span>
            @include grid-container
          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:text-no-wrap tablet:maxw-card-lg display-inline-flex">
          <span class="text-normal">
            .grid-<code>row</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex">
          <span>
            grid-<code>row</code>
          </span>
        </td>
        <td data-title="Example" class="display-inline-flex">
          <span>
            @include grid-row
          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:text-no-wrap tablet:maxw-card-lg display-inline-flex">
          <span class="text-normal">
            .grid-<code>col</code>
          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex">
          <span>
            grid-<code>col</code>(<a href="{{ site.baseurl }}/design-tokens/spacing-units/" class="token">units</a>)
          </span>
        </td>
        <td data-title="Example" class="display-inline-flex">
          <span>

>             @include grid-col(6)

Reviewer 2: I think it would be helpful to show a few examples here
```suggestion
            @include grid-col
            @include grid-col(6)
            @include grid-col("auto")
```

          </span>
        </td>
      </tr>
      <tr>
        <th scope="row" data-title="Utility" class="tablet:text-no-wrap tablet:maxw-card-lg display-inline-flex">
          <span class="text-normal">

>             .grid-<code>gap</code>

Reviewer 2: I think it would be helpful to document both the default `.grid-gap` utility class along with size variants
```suggestion
            .grid-gap<br/>
            .grid-gap-<code>gap size</code>
```

          </span>
        </th>
        <td data-title="Mixin" class="display-inline-flex">
          <span>

>             grid-<code>gap</code>(<a href="{{ site.baseurl }}/design-tokens/spacing-units/" class="token">units</a>)

Reviewer 2: ```suggestion
            grid-gap(<code>gap size</code>)
```

          </span>
        </td>
        <td data-title="Example" class="display-inline-flex">
          <span>

>             @include grid-gap("lg")

Reviewer 2: ```suggestion
            @include grid-gap
            @include grid-gap("lg")
```

Author: Resolved in f44acecb

          </span>
        </td>

>       </tr>

Author: Yes this was resolved in [9585a40](https://github.com/uswds/uswds-site/pull/2276/commits/9585a40768d9d66717ee11f3f6ae2b9689325353)

    </tbody>
  </table>
  {% include utilities/utility-mixin-using.html %}
</section>  

<section id="advanced-settings" class="padding-top-4">
  <h2 class="site-h2 margin-y-0">Advanced settings</h2>

  {% include utilities/responsive-variants.html %}
  No settings available.

>   <div>

Author: Including the `utilities/responsive-variants.html` allows for this entire section to display (sans the "no settings available")

![image](https://github.com/uswds/uswds-site/assets/25211650/4b0a1689-1893-4665-8aba-d44607222216)

I figured there was value in displaying information on utility breakpoints and which breakpoints are turned on by default, but it carries the responsive variant settings section along with it, which the display grid does not have any settings for.

Maybe we could break down `responsive-variants.html` into smaller more modular includes so that I can pull just the breakpoint table and not the responsive settings table?

>   <div>

Author: We could also add a conditional if statement to the `responsive-variants` page to only display utility variants if they exist.

```diff
+  {% if page.utilities %}

  <p class="utility-text margin-top-4">Set whether a utility family outputs with responsive variants by setting the value of <code>$[utility_family]-settings.responsive</code> in your <a href="{{ site.baseurl }}/documentation/settings">USWDS settings configuration</a>. When modifying settings stored as a map, undefined map elements will keep their default values.</p>

  <div class="example border-left-05 border-secondary-light padding-left-105 margin-top-2 margin-bottom-2 bg-white">
    <h4 class="font-lang-2xs margin-top-0 margin-bottom-05">Example</h4>
    <pre class=" font-mono-xs margin-0 padding-0 bg-transparent">
@use "uswds-core" with (
  $background-color-settings: (
    responsive: true
  )
);
    </pre>
  </div>

  <table class="usa-table--borderless site-table-responsive site-table-simple">
    <thead>
      <tr>
        <th scope="col" class="display-inline-flex">Utility variable</th>
        <th scope="col" class="tablet:maxw-card-lg display-inline-flex">Default</th>
      </tr>
    </thead>
    <tbody class="font-mono-2xs">
      {% for utility in page.utilities %}
        <tr>
          <th scope="row" data-title="Utility variable" class="tablet:text-no-wrap display-inline-flex">
            <span class="text-normal">
              ${{ utility.var }}-settings.responsive
            </span>
          </th>
          <td data-title="Default" class="tablet:maxw-card-lg display-inline-flex">
            <span>
              {% if utility.responsive %}<span class="output-true">true</span>{% else %}false{% endif %}
            </span>
          </td>
        </tr>
      {% endfor %}
    </tbody>
  </table>
+ {% endif %}

>   <div>

Reviewer: ## Suggestion (blocking)

To avoid any confusion we should only pull in the information that we need and is relevant to the user.

I've imported the table and lowered the opacity on content that isn't relevant, but is included from the partial. The difference is pretty significant.

![image](https://github.com/uswds/uswds-site/assets/3385219/7a4a7b9a-d281-4d2e-b36f-1c9a662cd6ee)

We should either:
1. Find a way to generate the table with only the things we need.
2. Leave this partial out and resolve this in another issue (depends on LOE to separate).

>   <div>

Author: I added conditional logic mentioned above to leave out the responsive output section if the utility does not offer any settings. There were no additional regressions in the other utility pages that use this include.

I also left the instruction above the table because it adds instructions on how to implement responsive utility classes.

Resolved in [7cfdb75](https://github.com/uswds/uswds-site/pull/2276/commits/7cfdb75a973c214208fb9d31c537573c86b9d351)
